clients/bootstrap-agent-client/src/lib.rs

```diff
@@ -26,6 +26,12 @@ progenitor::generate_api!(
         ImportExportPolicy = omicron_common::api::external::ImportExportPolicy,
         TypedUuidForRackInitKind = omicron_uuid_kinds::RackInitUuid,
         TypedUuidForRackResetKind = omicron_uuid_kinds::RackResetUuid,
+    },
+    convert = {
+        {
+            type = "string",
+            pattern = r"^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$",
```

Comment:
fascinating, til!

Is there a worry that people will forget to do it?

Reply:
I have several concerns about this approach but it seems least bad given that schemars decided `semver::Version` ought to be treated as a literal :(

If you don't add this, Progenitor generates a wrapper struct around a String. I don't think it matters _that_ much if you leave it as-is; either you use that, or you get annoyed enough with it that you realize you can do this.

Reply:
Oh actually, does schemars come with an impl for `semver::Version`? If so then maybe specifying the pattern again isn't required? not sure.

Reply:
Even moving the pattern into a const and using it here doesn't work; the macro wants a literal. (I think it's being directly deserialized into the relevant schemars struct but I didn't get that far into figuring out the Progenitor source.)

Reply:
Oh I bet that's because it accepts a string and not a `ParseWrapper<syn::Expr>`.

Reply:
Ah looks like it actually needs the `SchemaObject to compare at compile time against. https://github.com/oxidecomputer/progenitor/blob/4a8467a4660df4308890b690808ecb898f128cd1/progenitor-macro/src/lib.rs#L157

Seems unavoidable in that case, I think.